### ISSUE TITLE
[feature request] Reproducible F-Droid app

### ISSUE THREAD

**Issue by jindamvani** [cvzi/ScreenshotTile#324]
@cvzi If possible can you please make reproducible F-Droid app. For more information see [Towards a reproducible F-Droid.](https://f-droid.org/en/2023/01/15/towards-a-reproducible-fdroid.html)
* see also: 
** [Total number of apps using RB: 6 + 75 = 81. Last updated: 2023-03-04](https://gitlab.com/obfusk/fdroid-misc-scripts/-/blob/master/reproducible/overview.md)
** [Encourage new apps enable reproducible build](https://gitlab.com/fdroid/fdroiddata/-/issues/2816)

**Comment by cvzi** [comment]
Is this possible for "old" apps? The documentation only talkes about new apps, not apps that are already in Fdroid.

**Comment by jindamvani** [comment]
> The documentation only talkes about new apps, not apps that are already in Fdroid.

fdroid devs are making effort to make reproducible apps by default for all new apps.

**Comment by cvzi** [comment]
Do you happen to know if there is another app that has switched to reproducible builds? To see how exactly they did it.

**Comment by cvzi** [comment]
I am not sure yet if I want to do it or not. 
Autoupdates won't work, meaning I would have to open a pull request to the [](https://gitlab.com/fdroid/fdroiddata)

**Comment by licaon-kter** [comment]
Autoupdates should work as expected... what do you think otherwise? @cvzi

**Comment by cvzi** [comment]
@licaon-kter That's what I understood from: https://forum.f-droid.org/t/reproducible-f-droid-app-not-possible-for-published-apps/21530/5 

The whole process is a bit elusive to me though

**Comment by licaon-kter** [comment]
There are two issues, both feature the word "autoupdate" :)

So, first, are the new releases still auto added/updated in the F-Droid metadata when you Tag as usual? Yes

Second, are OLD users (auto)updated to the NEW version with the changed signer? No, as Android does not allow this

OLD users need to UNinstall and REinstall the app, once the repro builds are available.

Solution applied for now, human touch, eg.:
* https://github.com/mehrvarz/webcall-android -> change to repro builds, has an in app news channel for the used server that announced the signer change
* Wireguard, changed to repro builds, announced on https://lists.zx2c4.com/pipermail/wireguard/2023-April/008045.html and https://floss.social/@fdroidorg/110181469096544486
* Mastodon, changed away from repro :( https://floss.social/@fdroidorg/110334276768072552

**Comment by licaon-kter** [comment]
Yes, second has both, so old users still get updates, but then the burden is on you to manually add a new MR for each new version.
